python/grass/jupyter/interact_display.py

```diff
@@ -0,0 +1,229 @@
+    if os.path.exists(os.path.join(dbase, location)):
+        if not overwrite:
+            print(("Location <%s> already exists. Operation canceled.") % location)
+            try:
+                os.remove(tmp_gisrc)
+            except Exception:
+                pass
+            return
+        else:
+            print(("Location <%s> already exists and will be overwritten") % location)
+            shutil.rmtree(os.path.join(dbase, location))
+
+    stdin = None
+    kwargs = dict()
+
+    ps = gs.pipe_command(
+        "g.proj",
+        quiet=True,
+        flags="t",
+        epsg=epsg,
+        location=location,
+        stderr=subprocess.PIPE,
+        env=env,
+        **kwargs,
+    )
+    
+    error = ps.communicate(stdin)[1]
+    try:
+        os.remove(tmp_gisrc)
+    except Exception:
+        pass
+
+    if ps.returncode != 0 and error:
+        raise ScriptError(repr(error))
+
+    try:
+        fd = codecs.open(
+            os.path.join(dbase, location, "PERMANENT", "MYNAME"),
+            encoding="utf-8",
+            mode="w",
+        )
+        
+        fd.write(os.linesep)
+        fd.close()
+    except OSError as e:
+        raise ScriptError(repr(e))
+
+class GrassRendererInteractive():
+    """This class creates interative GRASS maps with folium"""
+
+    def __init__(self, width=400, height=400):
+        """This initiates an instance of GrassRenderer"""
+        self.width=width
+        self.height=height
+        
+        # Get the parent location/mapset
+        self._rc_original = self._get_rc(os.environ)
+        
+        # Create new environment for tmp WGS84 location
+        rcfile, self._env = gs.create_environment(self._rc_original["GISDBASE"], "temp_folium_WGS84", "PERMANENT")       
+        
+        # Location and mapset and region
+        create_location(self._rc_original["GISDBASE"], "temp_folium_WGS84", epsg="4326", overwrite=True, env=self._env)
+        
+        self._rc_tmp = self._get_rc(self._env) # Get tmp location/mapset
+        
+        self._extent = self._convert_extent(env=os.environ) # Get the extent of the original area in WGS84
+        
+        # Set region to match original region extent
+        gs.run_command("g.region",
+                       n=self._extent["North"],
+                       s=self._extent["South"],
+                       e=self._extent["East"],
+                       w=self._extent["West"],
+                       env=self._env
+                      )
+        
+        # Get Center of tmp GRASS region
+        center = gs.parse_command("g.region", flags="cg", env=self._env)
+        center = (float(center["center_northing"]), float(center["center_easting"]))
+
+        # Create Folium Map 
+        self.map = folium.Map(
+            width=self.width,
+            height=self.height,
+            location=center,
+            tiles="cartodbpositron",
+        )
+
+        
+    def _get_rc(self, env=None):
```

Comment:
Likely not needed.

Reply:
I'm not sure how else to get `GISDBASE` which is required for `create_location` and `create_environment`. It isn't stored in `os.environ`. I also think I need the original location and mapset for v.proj and I'm not sure how to get that besides in the rcfile.

Reply:
> I'm not sure how else to get `GISDBASE` which is required for `create_location` and `create_environment`.

Actually, that's another issue. You should probably use some temporary directory for this. You can use standard Python library to get a temporary directory. (`GISDBASE` is just a directory - a directory where locations ("projects") are.) In standard GRASS session created by the _grass_ command, there is one already created and can be used for things like this, but it needs to be used with some care and _grass.script.setup.init_ is not setting it up unlike the _grass_ command.

> It isn't stored in `os.environ`.

The equivalent of the system environment in GRASS is so called _GIS environment_ (a legacy name) accessible through _g.gisenv_ (_gs.gisenv_ function in Python). It is interface to the rc file (session file).

> I also think I need the original location and mapset for v.proj and I'm not sure how to get that besides in the rcfile.

Right, but the mapset should come from the _find_file_ call.